default/scripting/specials/ASTEROID_COATING.focs.txt

```diff
@@ -24,10 +24,16 @@ Special
                 OwnedBy affiliation = AnyEmpire
             ]     
             activation = Source
-            effects = [
-                AddSpecial name = "ASTEROID_COATING_OWNED_SPECIAL"
-                RemoveSpecial name = "ASTEROID_COATING_SPECIAL"
+            effects = SetStealth value = 5
```

Comment:
This sets the stealth of the object that has the special attached, which is not the same as having two specials with different stealth levels for the specials themselves, which was, I think, the main reason that there were two separate special definitions.

Reply:
we do not track stealth per special currently; a special "instance" is actually a `std::map<std::string, std::pair<int, float>> m_specials;` in UniverseObject.

Also the role of stealth of specials is not really well defined. E.g. ion storms should not be able to affect the stealth of a special.

so currently the implementation using two specials is the only option. not sure we should change something in the backend (e.g. adding stealth values to the m_specials map)

Reply:
alternative is to make special stealth a valueref.

Reply:
> alternative is to make special stealth a valueref.

so one could return in this case a different value based on ownership; clever